storage/innobase/xtrabackup/src/fil_cur.cc

```diff
@@ -102,6 +106,41 @@ const char *xb_get_relative_path(
   }
 }
 
+/** check for a tablespace in data dictonary
+@in tablespace_name database/tablespace name to check
+@in space_id 	space_id to check
+return true if found or false
+*/
+static bool check_for_tablespace_in_dd(std::string tablespace_name,
+                                       space_id_t space_id) {
+  static dd_tablespaces dd_space_name_id_map =
+      build_tablespace_space_id_map(mysql_connection);
```

Comment:
I dont like static variables inside function. Better to have them at file level scope.